Thuan2

thuan122 opened the pull request.

thuan122 committed:
absence request, absence type model, controller added

thuan122 committed:
Update User.php

thuan122 committed:
migration fixed

thuan122 committed:
Update AbsenceController.php

thuan122 committed:
Merge branch 'main' into thuan2

HoangMoli commented on routes/api.php:
```diff
@@ -23,6 +25,11 @@
+
     Route::post('permission/assign', [PermissionController::class, 'assignPermissionToRole']);
     Route::post('role/assign', [RoleController::class, 'assignRoleToUser']);
+    Route::post('user/salary', [UserController::class, 'updateSalary']);
```
should be put user/{id}/salary

HoangMoli commented on database/migrations/2023_05_11_040119_add_salary_to_users.php:
```diff
@@ -0,0 +1,32 @@
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->float('salary')->nullable(false)->default(0);
```
add this directly to users table

HoangMoli commented on database/migrations/2023_05_11_040127_create_absence_type_table.php:
```diff
@@ -0,0 +1,84 @@
+            $table->string('created_by')->nullable();
+            $table->string('updated_by')->nullable();
+            $table->boolean('is_active')->default(true);
+            $table->timestamp('created_at')->useCurrent();
```
should use the new CsutomBlueprint

HoangMoli commented on database/seeders/UserSeeder.php:
```diff
@@ -26,14 +27,24 @@ public function run()
             ]);
 
+        DB::table(AbsenceType::retrieveTableName())
+            ->where('id', '>', 3)
```
shoud be "not code W" and "now code W/2"

HoangMoli commented on app/Models/AbsenceType.php:
```diff
@@ -0,0 +1,91 @@
+        'description' => 'string'
+    ];
+
+    const DEFAULT_AMOUNT = 0;
```
default should be differed by each absence type

HoangMoli commented on app/Models/User.php:
```diff
@@ -64,19 +70,58 @@ class User extends Authenticatable
+        return $this->hasMany(AbsenceRequest::class);
+    }
+
+    public function updateSalary(Request $request)
```
comment

HoangMoli commented on app/Models/User.php:
```diff
@@ -64,19 +70,58 @@ class User extends Authenticatable
+            } else {
+                return Helper::getResponse('', 'Unauthorized');
+            }
+        } catch (\Throwable $th) {
```
should use common

HoangMoli commented on app/Models/User.php:
```diff
@@ -64,19 +70,58 @@ class User extends Authenticatable
+            ]
+        );
+
+        if ($validator->fails()) {
```
should use common

thuan122 committed:
update from main

thuan122 committed:
Update AuthController.php

thuan122 committed:
minor fixed

update salary code convention (still get missing user_id message when request)

add default_amount to ABSENCE_TYPES array

minor fixed to user seeder and createUser function

HoangMoli commented on database/migrations/2023_05_11_040153_create_absence_requests_table.php:
```diff
@@ -0,0 +1,46 @@
+            $table->enum('status', $status)->default($status[0]);
+            $table->foreign('absence_type_id')->references('id')->on(AbsenceType::retrieveTableName());
+            $table->foreign('user_id')->references('id')->on('users');
+
```
.

thuan122 committed:
migration minor fixed
